src/sdk/namespace_fs.js

```diff
@@ -2040,6 +2029,62 @@ class NamespaceFS {
     // INTERNALS //
     ///////////////
 
+    /**
+     * _get_restore_status is a pure function which consumes the xattr
+     * of the target file and returns its current state.
+     * @param {nb.NativeFSXattr} xattr xattr of the target file
+     * @param {Date} now - current time
+     * @param {string} [file] - file name for which the xattr is provided
+     * @returns {{ongoing?: boolean; state: nb.RestoreState; expiry_time?: Date}}
+     */
+    _get_restore_status(xattr, now, file = '') {
+        // Total 6 states (2x restore_request, 3x restore_expiry)
+        let restore_request;
+        let restore_expiry;
+
+        const restore_request_xattr = xattr[GlacierBackend.XATTR_RESTORE_REQUEST];
+        if (restore_request_xattr) {
+            const num = Number(restore_request_xattr);
+            if (num > 0 && Number.isInteger(num)) {
```

Comment:
The main question I have about this format is whether having a simple restore_request that only captures the number of days is enough. I see the restore request in S3 can have a lot of parameters and I expect we might need to include those in the request xattr as well (think for example on the Tier field which can be Standard | Bulk | Expedited). Can we think if it is worth storing the actual json of the requested request in this xattr instead of just the raw days? This will also allow us to evolve the format easier.

Reply:
Yeah, I don't see any issue with having JSON stored here. Our policy based solution on the other hand might need some updates to accommodate this change.

Reply:
We discussed few options for the request structure:

days float - e.g.
1
0.25

millis- e.g.
24*3600*1000

json - e.g.
{"days":1, "tier": "expedited" }
{"hours": 6}

For now we think days float is the simplest one to follow.

Reply:
Went for days as float for now. Resolving.